File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def solution(n):
    bn = bin(n)
    bn = bn[2:]
    num1 = bn.count('1')
    nn = n+1
    while 1:
        tn = bin(nn)
        tn = tn[2:]
        num2 = tn.count('1')
        if num2 == num1:
            break
        nn = nn + 1
    answer = nn
    return answer

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.java
class Solution {
    public int solution(int n) {
        int one = checkOne(n);
        int answer = n+1;
        
        while(true) {
            if(one == checkOne(answer)) return answer;
            answer++;
        }
    }
    
    public int checkOne(int n) {
        String s = Integer.toBinaryString(n);
        int count = 0;
        for(int i = 0; i< s.length(); i++) {
            if(s.charAt(i) == '1') {
                count++;
            }
        }
        return count;
    }
}